# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: modules/libpref/init/StaticPrefList.yaml
# Commit: 0e68db4ad6af
# Full Hash: 0e68db4ad6afe1f43d97533316d80e5aadd96082
# Author: Luca Greco <lgreco@mozilla.com>
# Date: 2020-07-01 21:32:12
# Regressor Bug: 1609920
# File Overlap Count: 2
# Description:
#   Bug 1609920 - part 1: Allow the WebExtension Framework to register a moz-extension service worker. r=dom-workers-and-storage-reviewers,asuth,mixedpuppy
#   
#   - Adds the new about:config pref "extensions.backgroundServiceWorker.enabled" (currently defaults to false).
#   - Adds the background.service_worker property to the manifest JSON schema definition
#   - Locks background.service_worker manifest property behind the new preference
# ==============================================================================

diff -r 2a107eabe00c -r 0e68db4ad6af modules/libpref/init/StaticPrefList.yaml
--- a/modules/libpref/init/StaticPrefList.yaml	Mon Jun 29 16:18:44 2020 +0000
+++ b/modules/libpref/init/StaticPrefList.yaml	Wed Jun 10 12:13:29 2020 +0000
@@ -3435,6 +3435,13 @@
   value: false
   mirror: always
 
+# Whether the background.service_worker in the extension manifest.json file
+# is enabled.
+- name: extensions.backgroundServiceWorker.enabled
+  type: bool
+  value: false
+  mirror: once 
+
 # This pref governs whether we enable content script CSP in extensions.
 - name: extensions.content_script_csp.enabled
   type: bool